airbyte-webapp/src/pages/ConnectionPage/pages/ConnectionItemPage/components/EnabledControl.tsx

```diff
@@ -25,47 +25,54 @@ const Content = styled.div`
   align-items: center;
 `;
 
-type IProps = {
+type EnabledControlProps = {
   connection: Connection;
   disabled?: boolean;
   frequencyText?: string;
 };
 
-const EnabledControl: React.FC<IProps> = ({ connection, disabled, frequencyText }) => {
+const EnabledControl: React.FC<EnabledControlProps> = ({ connection, disabled, frequencyText }) => {
   const { mutateAsync: updateConnection } = useUpdateConnection();
   const analyticsService = useAnalyticsService();
 
+  const [loading, setLoading] = useState(false);
+
   const onChangeStatus = async () => {
-    await updateConnection({
-      connectionId: connection.connectionId,
-      syncCatalog: connection.syncCatalog,
-      schedule: connection.schedule,
-      namespaceDefinition: connection.namespaceDefinition,
-      namespaceFormat: connection.namespaceFormat,
-      prefix: connection.prefix,
-      operations: connection.operations,
-      status: connection.status === ConnectionStatus.ACTIVE ? ConnectionStatus.INACTIVE : ConnectionStatus.ACTIVE,
-    });
+    setLoading(true);
+    try {
+      await updateConnection({
+        connectionId: connection.connectionId,
+        syncCatalog: connection.syncCatalog,
+        schedule: connection.schedule,
+        namespaceDefinition: connection.namespaceDefinition,
+        namespaceFormat: connection.namespaceFormat,
+        prefix: connection.prefix,
+        operations: connection.operations,
+        status: connection.status === ConnectionStatus.ACTIVE ? ConnectionStatus.INACTIVE : ConnectionStatus.ACTIVE,
+      });
 
-    analyticsService.track("Source - Action", {
-      action: connection.status === ConnectionStatus.ACTIVE ? "Disable connection" : "Reenable connection",
-      connector_source: connection.source?.sourceName,
-      connector_source_id: connection.source?.sourceDefinitionId,
-      connector_destination: connection.destination?.name,
-      connector_destination_definition_id: connection.destination?.destinationDefinitionId,
-      frequency: frequencyText,
-    });
+      analyticsService.track("Source - Action", {
+        action: connection.status === ConnectionStatus.ACTIVE ? "Disable connection" : "Reenable connection",
+        connector_source: connection.source?.sourceName,
+        connector_source_id: connection.source?.sourceDefinitionId,
+        connector_destination: connection.destination?.name,
+        connector_destination_definition_id: connection.destination?.destinationDefinitionId,
+        frequency: frequencyText,
+      });
+    } catch {}
```

Comment:
Could catch this error and render below. Maybe worth adding the logic into ToggleLabel?

Reply:
Since we're not really catching anything here, and I assume we just want to make sure we're getting back into the non loading state, could we instead just switch this empty `catch` into a `finally` to be more expressive about what we're actually trying to do here.

Reply:
There was chatter about error handling, but I have my concerns about it.
Switched to a `finally`